src/main/java/com/cefriel/template/TemplateExecutor.java

```diff
@@ -37,75 +36,102 @@
 public class TemplateExecutor {
 
 	private final org.slf4j.Logger log = LoggerFactory.getLogger(TemplateExecutor.class);
+	private Map<String, Reader> readers = null;
+	private final TemplateFunctions templateFunctions;
+	private final boolean failInvalidRef;
+	private final boolean trimTemplate;
+	private final boolean templateInResourcesFolder;
+	private final TemplateMap templateMap;
+	private final Formatter formatter;
+	private final VelocityContext velocityContext;
+	private final VelocityEngine velocityEngine;
+
+	public TemplateExecutor(Map<String, Reader> readers, TemplateFunctions templateFunctions, boolean failInvalidRef, boolean trimTemplate, boolean templateInResourcesFolder, TemplateMap templateMap, Formatter formatter) {
```

Comment:
Reader(s) should probably not be in the constructor but passed as parameter when the execution of a template is requested. Via CLI this is not important, but if used as a library can facilitate the reuse of the same TemplateExecutor instance.

Reply:
The problem is not initialising the VelocityEngine twice; this is why readers should not be in the constructor. It is ok to create different VelocityContext when requesting the VelocityEngine to execute a template.

https://github.com/cefriel/mapping-template/blob/e8ae166d7dd2b58995e69b5f954349656cdfb73a/src/main/java/com/cefriel/template/TemplateExecutor.java#L187